typedb/connection/client.py

```diff
@@ -32,23 +36,30 @@
 
 
 class _TypeDBClientImpl(TypeDBClient):
+    _PULSE_INTERVAL_SECONDS = 5
 
     # TODO: Detect number of available CPUs
     def __init__(self, address: str, parallelisation: int = 2):
         self._address = address
         self._transmitter = RequestTransmitter(parallelisation)
         self._sessions: Dict[bytes, _TypeDBSessionImpl] = {}
+        self._sessions_lock = Lock()
         self._is_open = True
+        self._pulse_scheduler = sched.scheduler(time.time, time.sleep)
+        self._pulse = self._pulse_scheduler.enter(delay=self._PULSE_INTERVAL_SECONDS, priority=1, action=self._transmit_pulses, argument=())
+        Thread(target=self._pulse_scheduler.run, name="session_pulse_{}".format(uuid4()), daemon=True).start()
```

Comment:
Schedule pulses to be transmitted every 5 seconds.

Pulses will be transmitted for all sessions that are open at the time of execution.